plugins/epochstorage/plugin.go

```diff
@@ -0,0 +1,474 @@
+	for _, i := range epochToRemove {
+		delete(epochContents, i)
+	}
+	epochContentsMutex.Unlock()
+
+	epochVotersWeightMutex.Lock()
+	for _, i := range epochToRemove {
+		delete(epochVotersWeight, i)
+	}
+	epochVotersWeightMutex.Unlock()
+
+	// update minEpochIndex
+	minEpochIndex = epochOrder[0]
+
+	committedEpochsMutex.Lock()
+	if len(committableEpochs) < maxEpochContentsToKeep {
+		committedEpochsMutex.Unlock()
+		return
+	}
+	committableEpochs = committableEpochs[len(committableEpochs)-maxEpochContentsToKeep:]
+	committedEpochsMutex.Unlock()
+}
+
+func GetCommittedEpochs() (ecRecords map[epoch.Index]*epoch.ECRecord) {
+	ecRecords = make(map[epoch.Index]*epoch.ECRecord, 0)
+
+	committedEpochsMutex.RLock()
+	for _, record := range committableEpochs {
+		ecRecords[record.EI()] = record
+	}
+	committedEpochsMutex.RUnlock()
+
+	return
+}
+
+func GetPendingBranchCount() map[epoch.Index]uint64 {
+	return deps.NotarizationMgr.PendingConflictsCountAll()
+}
+
+func GetEpochMessages(ei epoch.Index) ([]tangle.MessageID, error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []tangle.MessageID{}, err
+	}
+
+	var msgIDs []tangle.MessageID
+	stores.messageIDs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		var msgID tangle.MessageID
+		if _, err := msgID.Decode(key); err != nil {
+			panic("MessageID could not be parsed!")
+		}
+		msgIDs = append(msgIDs, msgID)
+		return true
+	})
+
+	return msgIDs, nil
+}
+
+func GetEpochTransactions(ei epoch.Index) ([]utxo.TransactionID, error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []utxo.TransactionID{}, err
+	}
+
+	var txIDs []utxo.TransactionID
+	stores.transactionIDs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		var txID utxo.TransactionID
+		if _, err := txID.Decode(key); err != nil {
+			panic("TransactionID could not be parsed!")
+		}
+		txIDs = append(txIDs, txID)
+		return true
+	})
+
+	return txIDs, nil
+}
+
+func GetEpochUTXOs(ei epoch.Index) (spent, created []utxo.OutputID, err error) {
+	stores, err := getEpochContentStorage(ei)
+	if err != nil {
+		return []utxo.OutputID{}, []utxo.OutputID{}, err
+	}
+
+	stores.createdOutputs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		outputID, err := outputIDFromBytes(key)
+		if err != nil {
+			panic(err)
+		}
+		created = append(created, outputID)
+
+		return true
+	})
+
+	stores.spentOutputs.IterateKeys(kvstore.EmptyPrefix, func(key kvstore.Key) bool {
+		outputID, err := outputIDFromBytes(key)
+		if err != nil {
+			panic(err)
+		}
+		spent = append(spent, outputID)
+
+		return true
+	})
+
+	return spent, created, nil
+}
+
+func GetEpochVotersWeight(ei epoch.Index) (weights map[epoch.ECR]map[identity.ID]float64) {
+	epochVotersWeightMutex.RLock()
+	defer epochVotersWeightMutex.RUnlock()
+	if _, ok := epochVotersWeight[ei]; !ok {
+		return
+	}
+
+	duplicate := make(map[epoch.ECR]map[identity.ID]float64, len(epochVotersWeight[ei]))
```

Comment:
You already have `weights` declared as your return variable. use make on that one